File: phase-2-web/frontend/types/notification.ts
export interface Notification {
  id: number;
  user_id: number;
  type: string;
  title: string;
  message: string;
  related_task_id: number | null;
  related_user_id: number | null;
  read: boolean;
  created_at: string;
}

export type NotificationType =
  | 'task_assigned'
  | 'task_due'
  | 'comment_added'
  | 'task_completed'
  | 'task_updated';

File: phase-2-web/backend/app/models/notification.py
"""
Notification Model

Stores in-app notifications for users.
"""

from datetime import datetime, timezone
from typing import Optional
from sqlmodel import Field, SQLModel


class Notification(SQLModel, table=True):
    """
    In-app notification for users.
    """
    __tablename__ = "notifications"

    id: Optional[int] = Field(default=None, primary_key=True)
    user_id: str = Field(foreign_key="users.id", index=True)

    # Notification content
    type: str = Field(max_length=50)  # task_assigned, task_due, comment_added, etc.
    title: str = Field(max_length=200)
    message: str = Field(max_length=500)

    # Related entity
    related_task_id: Optional[int] = Field(default=None)
    related_user_id: Optional[str] = Field(default=None)

    # Status
    read: bool = Field(default=False)

    # Metadata
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
